src/gui_common/charts/line/LineChart.cs

```diff
@@ -385,6 +395,9 @@ public void ClearDataSets()
 
             // Initialize line
             var dataLine = new DataLine((LineChartData)data.Value, data.Key == defaultDataSet, isChild ? 2 : 1);
+            dataLine.Connect("tree_exited", this, nameof(RemoveDataLineTooltipOnDeletion), new Array(dataLine));
```

Comment:
I think this should be more general, that is to connect `LineChart`'s own `tree_exited` signal to a new method called `ClearTooltips` to make sure everything else gets destroyed. The tooltip clearing logic in `ClearChart` needs to be moved there and the former just simply call `ClearTooltips`.

Reply:
LineChart delete lines when data is updated. (And that was the main cause of orphaned tooltips).

Reply:
Yes that's because `ClearChart` is called which among other things also clears the tooltips.